Update (#68)
OpenVoiceOS/community-docs: docs/qs_first_boot.md

Reviewer: Punctuation at the ends of lines

# OVOS Quick Start - First Boot

You now have a flashed boot medium with OVOS installed.  Now what?

Insert your boot medium, and power on your device.

> **NOTE** On a RPi4 with a USB device, it is recommended to use the top-right USB3 port.

Reviewer 2: ```suggestion
**NOTE** If using a Raspberry Pi 4 with a device conneected via USB-A, the top-right USB 3.2 Gen 1 port is recommended.
```

[Per the USB-IF, USB 3.0 is now USB 3.2 Gen 1.](https://www.diyphotography.net/usb-3-is-about-to-get-renamed-again-and-its-going-to-get-even-more-confusing/#:~:text=usb%203.0%2C%20which%20became%20usb%203.1%20gen%201%20is%20now%20usb%203.2%20gen%201) I know, we all hate it. We still need to use the official names though, surely?

By the way,

![This](https://assets.raspberrypi.com/static/blueprint-labelled-97975f4b1159239a8e248d180be87e3e.svg)

appears to demonstrate that the top-right port is actually the USB 2.0 port, whereas the middle port is the 3.2 Gen 1 port.

Each image has a different experence when you first boot your device.

[Buildroot - GUI](#Buildroot-GUI)

[Raspbian-ovos - Headless](#Raspbian-ovos-Headless)

**Notes on internet** While OVOS can run fully offline, with on device STT and TTS, your device will be lacking many of the things you expect from a smart assistant.  On a RPi, this includes the date and time because the Pi lacks a Real Time Clock, and therefor needs to connect to a server to set those on your device.

## Buildroot - GUI

When you first boot the Buildroot image, you will be greated with an OVOS splash screen as shown below.

> ![](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Buildroot%20bootsplash.png)

Reviewer 2: ```suggestion
![The splash screen is visible here.](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Buildroot%20bootsplash.png)
```

Slow internet connections exist, and does accessibility software. Always good to remember that.

> As this is the first time you have booted your device, it might take a bit longer than normal as the system is preparing its local filesystem and extending it over the full size of the sdcard/USB device.
> Eventually the progress bar will be filled up indicating the Operating System has been fully booted, after which the ovos-shell animated loading screen will be shown.

Reviewer 2: ```suggestion
As this is the first time you have booted your device, it might take a bit longer than normal as the system is preparing its local filesystem and extending it over the full size of the sdcard/USB device.

Eventually the progress bar will be filled up indicating the Operating System has been fully booted, after which the ovos-shell animated loading screen will be shown.
```

More readable. Unless using pre-line indentations (which [line numbers / editor-based indentation] render mostly obsolete, and which Markdown disallows anyway) paragraph spaces should be used instead of just a line-break.

Then again, that is unless the preview deceives me, and GitHub now puts paragraph spacing between lines, like the suggestion preview demonstrates:

![image](https://github.com/OpenVoiceOS/community-docs/assets/42837531/184b830a-4dfd-4342-820a-1518a290bc2b)

> ![](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Spinner.png)

Reviewer 2: ```suggestion
![The Loading Spinner is visible here.](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Spinner.png)
```

Again, since this is your first boot, this might take a bit longer as the ovos-core configuration is populated and skills are being setup.

#### Ethernet

> There should be no issues connecting automatically if your router accepts DHCP requests

Reviewer: ```suggestion
There should be no issues connecting automatically if your router accepts DHCP requests.
```

#### WiFi
If you do not have an internet connection, you will be prompted with a screen with options to connect.

> ![](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Network%20setup.png?raw=true)

Reviewer 2: ```suggestion
![The image is visible here.](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Network%20setup.png?raw=true)
```

You can also skip this step to configure it later or never ask it again if you want your device to run fully offline. (Bear in mind you need to configure your device to use local STT and TTS engines and obvious asking your device things that require internet will not work.  This includes the date and time as a Raspberry Pi does not have a Real Time Clock and therefore does not know this data without being told)

**On Device Setup**

If this option is selected the next screen will have a list of avaliable WiFi connections

> ![](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Select%20wifi%20on-device.png?raw=true)

Reviewer 2: ```suggestion
![The image is visible here.](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Select%20wifi%20on-device.png?raw=true)
```

Select your desired network, and enter a password if needed

> ![](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Insert%20wifi%20password.png?raw=true)

Reviewer 2: ```suggestion
![The image is visible here.](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Insert%20wifi%20password.png?raw=true)
```

If everything went correctly, you should be connected to the internet and after a short period of time, OVOS is loading skills that require internet, you will be presented with the homescreen.

> ![](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Horizontal%20homescreen.jpg?raw=true)

Reviewer 2: ```suggestion
![The image is visible here.](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Horizontal%20homescreen.jpg?raw=true)
```

**Configure WiFi after skipping intial setup**

If you have skipped the Wi-Fi network setup before, or you just want to reconfigure your network. On the homescreen of your OpenVoiceOS device swipe down the top menu and click the "Wi-Fi" icon. This brings you to the same on-device configuration screen.

> ![](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Network%20connections.png)

Reviewer 2: ```suggestion
![The image is visible here.](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Network%20connections.png)
```

From here you can select another network or click the configuration icon on the right of connected network for details or to remove it from the configured networks.

> ![](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Connection%20details.png)

Reviewer 2: ```suggestion
![The image is visible here.](https://raw.githubusercontent.com/OpenVoiceOS/ovos_assets/master/Images/Screenshot%20-%20Connection%20details.png)
```

**Mobile Setup**

If this option is chosen, you will be prompted to connect your mobile device, or computer to the hotspot `OVOS`

> ![](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Connect%20hotspot.png?raw=true)

Reviewer 2: ```suggestion
![The image is visible here.](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/Screenshot%20-%20Connect%20hotspot.png?raw=true)
```

The rest of this option coinsides with the headless image WiFi setup.  [Continue Mobile WiFi Setup](#WiFi-1)

## Raspbian-ovos - Headless
If you are trying to run OVOS on a RPi3, this is the image to use.  It **DOES NOT** provide a GUI, therefore some things that are avaliable on the buildroot image, are not avaliable here.

Once again, it may take several minutes for this first boot to complete.  OVOS resizes the partition on the drive, and loads the required plugins.

#### Ethernet
There should be no issues connecting automatically if your router accepts DHCP requests

#### WiFi
On first boot, you should hear a voice prompt instructing you to connect to the `OVOS` hotspot

> "Open a browser to 'start dot openvoiceos dot com'"

> Connect your mobile device or computer to the `OVOS` hotspot and open the webpage http://start.openvoiceos.com **NOTE** This is NOT the official OVOS website, rather a "Hotspot" created by OVOS and will be removed after the device is connected to WiFi

Reviewer 2: ```suggestion
Connect your mobile device or computer to the `OVOS` hotspot and open the webpage [start.openvoiceos.org](https://start.openvoiceos.org/). **NOTE** This is *not* the official OVOS website, instead a "Hotspot" created by OVOS and will be removed after the device is connected to WiFi.
```

I also have a tendency to use `http` instead of `https` for sites that support `http` since it allows older clients to connect without issue, but for sites that auto-redirect to `https` anyway, it slows the connection process.

Additionally,

> This is *not* the official OVOS website, instead a "Hotspot"

doesn't explain what it is. What's a hotspot? A user should probably be informed of this.

However, https://start.openvoiceos.org/ redirects me to https://openvoiceos.org/ now, so is that paragraph even necessary?

Author: I believe it IS necessary because the hotspot that is created does not go to that site.  it is `http://start.openvoiceos.com`  Note the `.com` not `.org`

But I did put a note on what a "HotSpot" is

Choose from the list of WiFi access points from the list on your mobile device.

> ![](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/iPhone%20-%20WiFi%20setup%20startpage.PNG?raw=true)

Reviewer 2: ```suggestion
![The image is visible here.](https://github.com/OpenVoiceOS/ovos_assets/blob/master/Images/iPhone%20-%20WiFi%20setup%20startpage.PNG?raw=true)
```
